backend/app/alembic/versions/dc391733e10a_add_extent_to_gerrydb.py

```diff
@@ -0,0 +1,85 @@
+"""add extent to gerrydb
+
+Revision ID: dc391733e10a
+Revises: 5ab466c5650a
+Create Date: 2024-10-27 19:38:13.798056
+
+"""
+
+from typing import Sequence, Union
+
+from alembic import op
+import sqlalchemy as sa
+from app.models import DistrictrMap
+
+# revision identifiers, used by Alembic.
+revision: str = "dc391733e10a"
+down_revision: Union[str, None] = "5ab466c5650a"
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on = ("167892041d95",)  # geometry col renaming
+down_revision = "167892041d95"
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    # Make column nullable to avoid errors during initial insert
+    op.add_column(
+        "districtrmap", sa.Column("extent", sa.ARRAY(sa.Float()), nullable=True)
+    )
+
+    # Use a session to interact with the database
+    bind = op.get_bind()
+    Session = sa.orm.sessionmaker(bind=bind)
+    session = Session()
+
+    # retreive table names from the districtrmap parent_layer field
+    table_names = session.execute(sa.select(DistrictrMap.parent_layer)).scalars().all()
+
+    if not table_names:
+        raise ValueError("No matching table names found in the index table")
+
+    cases = []
+    for table_name in table_names:
+        case = f"""
+        WHEN districtrmap."parent_layer" = '{table_name}' THEN (
+            SELECT
+                ARRAY[
+                    ST_XMin(ST_Extent(ST_Transform(geometry, 4326))),
+                    ST_YMin(ST_Extent(ST_Transform(geometry, 4326))),
+                    ST_XMax(ST_Extent(ST_Transform(geometry, 4326))),
+                    ST_YMax(ST_Extent(ST_Transform(geometry, 4326)))
+                ]
+            FROM gerrydb."{table_name}", public.districtrmap
+            
+        )
+        """
+
+        cases.append(case)
+    # Combine all cases into a single SQL statement
+    case_statement = " ".join(cases)
+    # Execute a single UPDATE statement
+    update_query = sa.text(
+        f"""
+        UPDATE districtrmap
+        SET extent = CASE
+            {case_statement}
+        -- ELSE ARRAY[0, 0, 0, 0] if this fails, there is no
```

Comment:
PP: `ELSE NULL` – and we default to Kansas view? Non-blocking esp. since this is in a migration.

Reply:
agreed and i don't expect it'll happen on prod. the column is set to not allow for nulls and I'd rather keep it that way so will leave as-is for now, but am open to better handling issues down the line

Reply:
Oh I thought I saw `nullable=True` in the model, though now that I'm thinking about it not in the revision.